cmd/source_profile.go

```diff
@@ -0,0 +1,283 @@
+		dydb.awsRegion = awsRegion
+	}
+	if dydbEndpoint, ok := params["dydbEndpoint"]; ok {
+		dydb.dydbEndpoint = dydbEndpoint
+	}
+	if schemaSampleSize, ok := params["schemaSampleSize"]; ok {
+		schemaSampleSizeInt, err := strconv.Atoi(schemaSampleSize)
+		if err != nil {
+			return dydb, fmt.Errorf("could not parse schemaSampleSize = %v as a valid int64", schemaSampleSize)
+		}
+		dydb.schemaSampleSize = int64(schemaSampleSizeInt)
+	}
+	return dydb, nil
+}
+
+type SourceProfileConnection struct {
+	ty    SourceProfileConnectionType
+	mysql SourceProfileConnectionMySQL
+	pg    SourceProfileConnectionPostgreSQL
+	dydb  SourceProfileConnectionDynamoDB
+}
+
+func NewSourceProfileConnection(source string, params map[string]string) (SourceProfileConnection, error) {
+	conn := SourceProfileConnection{}
+	switch strings.ToLower(source) {
+	case "mysql":
+		{
+			conn.ty = SourceProfileConnectionTypeMySQL
+			conn.mysql = NewSourceProfileConnectionMySQL(params)
+		}
+	case "postgresql", "postgres", "pg":
+		{
+			conn.ty = SourceProfileConnectionTypePostgreSQL
+			conn.pg = NewSourceProfileConnectionPostgreSQL(params)
+		}
+	case "dynamodb":
+		{
+			conn.ty = SourceProfileConnectionTypeDynamoDB
+			dydb, err := NewSourceProfileConnectionDynamoDB(params)
+			conn.dydb = dydb
+			if err != nil {
+				return conn, err
+			}
+		}
+	default:
+		return conn, fmt.Errorf("please specify a valid source database using -source flag, received source = %v", source)
+	}
+	return conn, nil
+}
+
+type SourceProfileConfig struct {
+	path string
+}
+
+func NewSourceProfileConfig(path string) SourceProfileConfig {
+	return SourceProfileConfig{path: path}
+}
+
+type SourceProfile struct {
+	ty     SourceProfileType
+	file   SourceProfileFile
+	conn   SourceProfileConnection
+	config SourceProfileConfig
+}
+
+func (src SourceProfile) ToLegacyDriver(source string) (string, error) {
+	switch src.ty {
+	case SourceProfileTypeFile:
+		{
+			switch strings.ToLower(source) {
+			case "mysql":
+				return "mysqldump", nil
+			case "postgresql", "postgres", "pg":
+				return "pg_dump", nil
+			case "dynamodb":
+				return "", fmt.Errorf("dump files are not supported with DynamoDB")
+			default:
+				return "", fmt.Errorf("please specify a valid source database using -source flag, received source = %v", source)
+			}
+		}
+	case SourceProfileTypeConnection:
+		{
+			switch strings.ToLower(source) {
+			case "mysql":
+				return "mysql", nil
+			case "postgresql", "postgres", "pg":
+				return "postgres", nil
+			case "dynamodb":
+				return "dynamodb", nil
+			default:
+				return "", fmt.Errorf("please specify a valid source database using -source flag, received source = %v", source)
+			}
+		}
+	case SourceProfileTypeConfig:
+		return "", fmt.Errorf("specifying source-profile using config not implemented")
+	default:
+		return "", fmt.Errorf("invalid source-profile, could not infer type")
+	}
+}
+
+// Source profile is passed as a list of key value pairs on the command line.
```

Comment:
move this comment to the top of toLegacyDriver

Reply:
Umm, I thought we should add a separate comment to ToLegacyDriver based on the code review comment, which I have done now. ToLegacyDriver will get deprecated soon, and thus I was wondering if we should just keep this comment here.